### PR TITLE
Enhanced nccl tests slurm validation playbook.

### PR DESCRIPTION
**Pull request by avolkov1**
Ability to specify any base container and dynamically compile a new enroot container with NCCL tests. Then runs the NCCL tests using the compiled enroot container.

### CODE REVIEW

File: playbooks/slurm-cluster/slurm-validation.yml [hunk 1]
```diff
@@ -1,33 +1,212 @@
 ---
-# Playbook designed to run a NCCL test across all nodes in a cluster of DGX-1s
+# Playbook designed to run a NCCL test across nodes in a cluster of DGXs
+# Example to run on NSL-B with two nodes:
```
Comment (supertetelman) [inline playbooks/slurm-cluster/slurm-validation.yml:3]: We shouldn't have references to internal clusters if it can be avoided.

Comment (avolkov1) [reply playbooks/slurm-cluster/slurm-validation.yml:3]: I can reword to:
```
# Example to run on two nodes with enp set for out-of-band NCCL init, no UCX and no HCOLL:
```

Or please suggest how to reword. This example will work in clusters without RDMA (no RoCE or infiniband). I want to document in the comments how to run on such clusters. Sometimes without the RDMA setup it is useful to test such cases.

Comment (avolkov1) [reply playbooks/slurm-cluster/slurm-validation.yml]: Fixed in commit af08cba.

**Review by ajdecon — APPROVED** [review]
LGTM. Two minor nit-picks mentioned inline but I don't think those are worth blocking this.

**Pull request merged**